src/app/reporting/reporting.cpp

```diff
@@ -66,7 +66,13 @@ using namespace chip;
 
 #define NULL_INDEX 0xFF
 
-static void conditionallySendReport(EndpointId endpoint, ClusterId clusterId);
+/*
+ * This method sends the report to the provided destination node at destinationEndpoint.
+ * If the reportDestination is kUndefinedNodeId, the method uses binding table to identify
+ * destination nodes and endpoints
+ */
+static void conditionallySendReport(EndpointId endpoint, ClusterId clusterId, NodeId reportDestination,
```

Comment:
According to the comment, It is confusing that the code works this way. Can it take a `MessageSendDestination` argument instead of `NodeId reportDestination`

Reply:
@pan-apple?